routemaster/state_machine/actions.py

```diff
@@ -36,6 +37,12 @@ def process_action(
     Returns whether the label progressed in the state machine, for which `True`
     implies further progression should be attempted.
     """
+    if not isinstance(state, Action):  # pragma: no branch
+        raise RuntimeError(  # pragma: no cover
```

Comment:
Surely this is a `ValueError`?

Reply:
Also in `process_gate` and `labels_needing_metadata_update_retry_in_gate`.

Reply:
I went with RuntimeError because I was seeing this as an invariant breaking somewhere, but I guess that this might be called directly in some other way, so maybe ValueError is better...

Reply:
b7b310e8ee469bb85ced98188a6274f0e935d324